aquarius/events/processors.py

```diff
@@ -353,3 +354,28 @@ def process(self):
         self.es_instance.update(self.asset, self.did)
 
         return self.asset
+
+
+class MetadataStateProcessor(EventProcessor):
+    def process(self):
+        self.did = make_did(self.event.address, self._chain_id)
+
+        try:
+            self.asset = self._es_instance.read(self.did)
+        except Exception:
+            self.asset = None
+
+        if self.event.args.state == MetadataStates.ACTIVE:
+            event_processor = MetadataCreatedProcessor(
+                self.event,
+                self.dt_contract,
+                self.sender_address,
+                self._es_instance,
+                self._web3,
+                self.allowed_publishers,
+                self.purgatory,
+                self._chain_id,
+            )
+            return event_processor.process()
+        elif self.asset:
```

Comment:
As Maria and David already know, this is my favourite trick and I always find myself suggesting it in code reviews: https://kaolin.tech/2016/10/untangling-conditionals-with-if-inversion/ Take time to read it, because I reference it a lot 😄 

Then after reading through that, please move the simpler block first (if state != active, delete and return True). Then you won't need an elif. You'll also retrieve the asset only if you need to delete it.